File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/recyclerview/VenueViewItem.java
package home.oleg.placesnearme.core_presentation.recyclerview;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import home.oleg.placesnearme.core_presentation.viewdata.VenueViewData;

public class VenueViewItem implements ItemViewType {

    public static int VIEW_TYPE = 1;

    private final VenueViewData venueViewData;

    public VenueViewItem(VenueViewData venueViewData) {
        this.venueViewData = venueViewData;
    }

    public VenueViewData getVenueViewData() {
        return venueViewData;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE;
    }

    public static List<VenueViewItem> map(List<VenueViewData> venueViewDataList) {
        List<VenueViewItem> venueViewItems = new ArrayList<>();

        for (VenueViewData venueViewData : venueViewDataList) {
            venueViewItems.add(new VenueViewItem(venueViewData));
        }

        return venueViewItems;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        VenueViewItem that = (VenueViewItem) o;
        return Objects.equals(venueViewData, that.venueViewData);
    }

    @Override
    public int hashCode() {

        return Objects.hash(venueViewData);
    }
}

File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/recyclerview/VenueViewItem.kt
package home.oleg.placesnearme.core_presentation.recyclerview

import home.oleg.placesnearme.core_presentation.viewdata.VenueViewData
import java.util.*

data class VenueViewItem(val venueViewData: VenueViewData) : com.smedialink.common.recyclerview.ItemViewType {

    override fun getViewType(): Int {
        return VIEW_TYPE
    }

    companion object {

        var VIEW_TYPE = 1

        fun map(venueViewDataList: List<VenueViewData>): List<VenueViewItem> {
            val venueViewItems = ArrayList<VenueViewItem>()

            for (venueViewData in venueViewDataList) {
                venueViewItems.add(VenueViewItem(venueViewData))
            }

            return venueViewItems
        }
    }
}
